hack/kicbase_version/release_kicbase_version.go

```diff
@@ -0,0 +1,230 @@
+  - DOCKER_TOKEN=<string>: Docker personal access token or password
+
+  @GitHub (ref: https://docs.github.com/en/free-pro-team@latest/packages/using-github-packages-with-your-projects-ecosystem/configuring-docker-for-use-with-github-packages)
+  - GITHUB_USERNAME=<string>: GitHub username
+  - GITHUB_TOKEN=<string>: GitHub [personal] access token
+*/
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+)
+
+const (
+	// default context timeout
+	cxTimeout = 600 * time.Second
+)
+
+var (
+	kicFile      = "../../pkg/drivers/kic/types.go"
+	kicVersionRE = `Version = "(.*)"`
+
+	// keep list of registries in sync with those in kicFile
+	registries = []registry{
+		{
+			name:     "Google Cloud Container Registry",
+			image:    "gcr.io/k8s-minikube/kicbase",
+			username: os.Getenv("GCR_USERNAME"),
+			password: os.Getenv("GCR_TOKEN"),
+		},
+		{
+			name:     "Docker Hub Container Registry",
+			image:    "docker.io/kicbase/stable",
+			username: os.Getenv("DOCKER_USERNAME"),
+			password: os.Getenv("DOCKER_TOKEN"),
+		},
+		{
+			name:     "GitHub Packages Registry",
+			image:    "docker.pkg.github.com/kubernetes/minikube/kicbase",
+			username: os.Getenv("GITHUB_USERNAME"),
+			password: os.Getenv("GITHUB_TOKEN"),
+		},
+	}
+
+	run = func(cmd *exec.Cmd, stdin io.Reader) error {
+		cmd.Stdin = stdin
+		var out bytes.Buffer
+		cmd.Stderr = &out
+		if err := cmd.Run(); err != nil {
+			return errors.Errorf("%s: %s", err.Error(), out.String())
+		}
+		return nil
+	}
+)
+
+// container registry name, image path, credentials, and updated flag
+type registry struct {
+	name     string
+	image    string
+	username string
+	password string
+	updated  bool
+}
+
+func (r *registry) setUpdated(updated bool) {
+	r.updated = updated
+}
+
+func main() {
+	ctx, cancel := context.WithTimeout(context.Background(), cxTimeout)
+	defer cancel()
+
+	// write log statements to stderr instead of to files
+	if err := flag.Set("logtostderr", "true"); err != nil {
+		fmt.Printf("Error setting 'logtostderr' glog flag: %v", err)
+	}
+	flag.Parse()
+	defer glog.Flush()
+
+	// determine current kic base image version
+	vCurrent, err := getKICVersion()
+	if err != nil {
+		glog.Fatalf("failed getting current kic base image version: %v", err)
+	}
+	if len(vCurrent) == 0 {
+		glog.Fatalf("cannot determine current kic base image version")
+	}
+	glog.Infof("current kic base image version: %s", vCurrent)
+
+	// determine release kic base image version
+	vRelease := strings.Split(vCurrent, "-")[0]
+	glog.Infof("release kic base image version: %s", vRelease)
+
+	// prepare local kic base image
+	image, err := prepareImage(ctx, vCurrent, vRelease)
+	if err != nil {
+		glog.Fatalf("failed preparing local kic base reference image: %v", err)
+	}
+	glog.Infof("local kic base reference image: %s", image)
```

Comment:
Could you please update this from glog to klog? We recently transitioned to this fork. It's basically a search & replace.

Reply:
@tstromberg i've updated it from glog to klog